lib/robots/viomi/ViomiValetudoRobot.js

```diff
@@ -381,6 +391,10 @@ class ViomiValetudoRobot extends MiioValetudoRobot {
             }));
         }
 
+        if (data["cur_mapid"] !== undefined) {
```

Comment:
I'm not a fan of this.
With this, we're at two dynamic viomi properties :/

Reply:
For some reason `get_props` with the `cur_mapid` is also slow on my robot. Is the mapId contained in the uploaded map that is parsed by the ViomiMapParser?

Reply:
I think I found it in my map dump. The mapId starts at offset 0x04 and is four byte long.

Reply:
In that case, you can add that to the metaData field of the ValetudoMap returned by the map parser and access that via the robot

plus this check to fail when the map is missing

`if (this.robot.state.map?.metaData?.defaultMap !== true) {`

Reply:
Done. I had to [modify the ValetudoMap to take the metaData option](https://github.com/Hypfer/Valetudo/pull/766/files#diff-40a906408dc9f223acf527c14e56488a5a1e4d7a26fff75f0585ff97f4ed8a3eL34).